internal/store/organizations.go

```diff
@@ -55,13 +56,15 @@ func (s *Store) Organizations(opts *atlas.OrganizationsListOptions) (*atlas.Orga
 }
 
 // Organization encapsulate the logic to manage different cloud providers.
-func (s *Store) Organization(id string) (*atlas.Organization, error) {
+func (s *Store) Organization(id string) (*v1alpha.ApiOrganizationView, error) {
 	switch s.service {
 	case config.CloudService, config.CloudGovService:
-		result, _, err := s.client.(*atlas.Client).Organizations.Get(s.ctx, id)
+		result, res, err := s.clientv2.OrganizationsApi.GetOrganization(s.ctx, id).Execute()
+		defer res.Body.Close()
 		return result, err
 	case config.CloudManagerService, config.OpsManagerService:
-		result, _, err := s.client.(*opsmngr.Client).Organizations.Get(s.ctx, id)
+		result, res, err := s.clientv2.OrganizationsApi.GetOrganization(s.ctx, id).Execute()
+		defer res.Body.Close()
```

Comment:
please rollback these lines, don't change cloud manager/ops manager calls APIs are different, configuration is different this may introduce big side effects

Reply:
Would it mean that we will need to supply two models to store?

Reply:
or an `interface{}` not the best use of strong typing from go but this is what we've done before in cases where they were completely different models

Another goal for a different project, backport the generated common APIs to the ops manager client, the opsmanager client "inherits" from the Atlas clients and overrides where needed but ops manager will never have an openapi spec so any change there will need to be manual

Reply:
check the projects store https://github.com/mongodb/mongodb-atlas-cli/blob/17c025603e81bed15935b3b55c23ed7bc2b89951/internal/store/projects.go#L70 they are completely different models between ops manager and atlas and yet it works

Reply:
That would be at the cost of the casting for request and response objects. 
I have done an investigation and I feel that we have options:

### 1. Use atlasv2 models in arguments and responses. 
Cast and change those atlasv2 models to old models inside store interface to support ops manager. 

#### Benefits

- We can auto-discover new features
- Atlas CLI will be benefiting from versioned API (new versions of models can be used on demand
- Separation/depreciation of the old client will be easier and would not require much engineering

### 2. Keep v1 models and transpose them to the v2 objects fo atlas cli

Technically it will mean less work if we keep interfaces as they are now and transpose them inside the method 
to the v2 format. However in situations when a field will be missing in v1 and required in v2 we will end up 

### 3. Weak store interface and model casting depending on the CLI type

Cli can create or operate with different models depending on the Atlas/OpsManager being present. 

### 4. Split store interface and create store depending on the CLI type

Since we have different cli instances that will be using different clients it actually makes sense to split the store into two different implementations:

- OpsManager Store (that will be using v1 client)
- Atlas Store (that will be using v1 client initially and migrate to v2 client by individual commands)

We will need to introduce another abstraction for each command that will create a required object for the store. 

@swattyT @gssbzn Sorry for the long thread.  It will probably make sense to move this thread to spec doc at some point.

Reply:
Projects are different between ops manager and atlas, here's what we've done bere https://github.com/mongodb/mongodb-atlas-cli/blob/17c025603e81bed15935b3b55c23ed7bc2b89951/internal/store/projects.go, not saying this is the best way to do it but it's an example, feel free to be opionated here but some word of advice

1. won't work becuase there are cases (like projects) where the structs are different between ops manager and atlas, different fields and options

2. not a fan since removing v1 will be harder but open to it

3. This is what we've done before

4. this may add some overhead, the store was built for the same case we are disucssing here having a way to let commands call the "store" and nto have to worry if it's ops manager or atlas

Reply:
I was considering option no3 or no4. It seems that we both agree with 3 is the way to go (and it is implemented in this PR already.

Reply:
As variation of the no4 we can use generics or have 2 conditional return types. 
I did play with a generic-based store and I found it quite tricky.
Finally returning 2 different types depending on the cli seemed like the best choice:

- it doesn't require casting 
- we can instantly look for the type we need.

Reply:
generics are new and I haven't had a time to play much with them so no strong opinions on using them